Framework/PythonInterface/plugins/algorithms/VesuvioCorrections.py

```diff
@@ -428,23 +544,43 @@ def _ms_correction(self):
                 sigma_x = float(params_dict[sigma_x_prop])
                 sigma_y = float(params_dict[sigma_y_prop])
                 sigma_z = float(params_dict[sigma_z_prop])
-                width = math.sqrt((sigma_x**2 + sigma_y**2 + sigma_z**2) / 3.0)
+                width = math.sqrt((sigma_x ** 2 + sigma_y ** 2 + sigma_z ** 2) / 3.0)
             else:
+                i = i + 1
                 continue
 
             atom_props.append(mass)
-            atom_props.append(intentisy)
+            atom_props.append(intensity)
             atom_props.append(width)
-            intensities.append(intentisy)
+            intensities.append(intensity)
+
+            # Check for NoneType is necessary as hydrogen constraints are
+            # stored in a C++ PropertyManager object, not a dict; call to
+            # __contains__ must match the C++ signature.
+            if self._back_scattering and symbol is not None and symbol in self._hydrogen_constraints:
+                self._hydrogen_constraints[symbol].value['intensity'] = intensity
+
+            i = i + 1
+
+        if self._back_scattering and contains_hydrogen:
+            material_builder = MaterialBuilder()
+            hydrogen = material_builder.setFormula('H').build()
+            hydrogen_intensity = \
+                self._calculate_hydrogen_intensity(hydrogen, self._hydrogen_constraints)
+            hydrogen_width = 5
```

Comment:
Should we add a property to be able to set this and set the default to `5`? It will be easier to experiment that way.